mcstatus/status_response.py

```diff
@@ -0,0 +1,662 @@
+
+        sample = [JavaStatusPlayer.build(player) for player in raw.get("sample", [])]
+        if len(sample) == 0:
+            sample = None
+        return cls(
+            online=raw["online"],
+            max=raw["max"],
+            sample=sample,
+        )
+
+
+@dataclass
+class BedrockStatusPlayers(BaseStatusPlayers):
+    """Class which represents `BedrockStatusResponse` players field."""
+
+
+@dataclass
+class JavaStatusPlayer:
+    """Class for storing player info. Only for Java."""
+
+    name: str
+    uuid: str
+
+    @classmethod
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build `JavaStatusPlayer` from raw response dict.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (name, uuid) are not present.
+        :raise TypeError: If the required keys (name - str, uuid - str) are not of the specified type.
+        :return: `JavaStatusPlayer` object.
+        """
+        _validate_data(raw, "player", [("name", str), ("id", str)])
+        return cls(name=raw["name"], uuid=raw["id"])
+
+
+@dataclass
+class BaseStatusVersion(ABC):
+    """Class for storing version info.
+
+    :param name: Version name. Example `1.18.0`.
+    :param protocol: Version protocol. See https://minecraft.fandom.com/wiki/Protocol_version.
+    """
+
+    name: str
+    protocol: int
+
+
+@dataclass
+class JavaStatusVersion(BaseStatusVersion):
+    """Class for storing Java version info."""
+
+    @classmethod
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build `JavaStatusVersion` from raw response dict.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (name, protocol) are not present.
+        :raise TypeError: If the required keys (name - str, protocol - int) are not of the specified type.
+        :return: `JavaStatusVersion` object.
+        """
+        _validate_data(raw, "version", [("name", str), ("protocol", int)])
+        return cls(name=raw["name"], protocol=raw["protocol"])
+
+
+@dataclass
+class BedrockStatusVersion(BaseStatusVersion):
+    """Class for storing Bedrock version info.
+
+    :param brand: Like `MCPE` or another.
+    """
+
+    brand: str
+
+
+def custom_eq(self, other) -> bool:
+    """Custom equality function for outdated subclasses."""
+    to_check = list(
+        (
+            (getattr(self, self_field.name) == getattr(other, other_field.name))
+            if self_field.compare and other_field.compare and (self_field.name == other_field.name)
+            else True
+        )
+        for self_field in fields(self)
+        for other_field in fields(other)
+    )
+    to_check.append(isinstance(self, other.__class__))
+    return all(to_check)
+
+
+_OLD_JAVA_INIT_SIGNATURE = Signature(
+    parameters=[
+        Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any]),
+    ]
+)
+
+
+class JavaStatusResponse(NewJavaStatusResponse):
```

Comment:
I actually discussed it with ItsDrike [here](https://github.com/py-mine/mcstatus/pull/306#discussion_r872993704).

Reply:
Looks like @ItsDrike agreed with me in his initial comment too, but also proposed a `Deprecated` prefix that I disagree with for the same reason. Classes should not have something in their name unless that something relates to the thing they are representing. Using `New` to signify that a class is new to the codebase is not a good naming practice. If there were to be a breaking change introduced the approach would be to create `v1` and `v2` folders that contain different versions of the class. Thus, allowing the name to be reused. However, this is not a breaking change and is only a "refactor", right? We should not be worried about replacing the old class here, since all the previous field names match, no?

Reply:
Did you read my message from link? I wrote there

> Classes prefixed with `New` **NOT A PART** of a public API (because not listed in `__all__`), so no-one should use it. There is no breaking changes, if you delete variable with `__` prefix.

This actually answers on your message. Yes, I agree classes should be named like something, that they represent. But `New` classes **NOT A PART** of a public API, and created just for easily removing deprecations after 2022-08.

Reply:
(I actually would like prefix them with `__` instead of `New`)

Reply:
Again, this does not look like a breaking change. This PR has "refactor" in the title. We can simply start using the new one because all the data contained in it is the same, no?

Refactor - To [rewrite](https://www.wordnik.com/words/rewrite) existing [source code](https://www.wordnik.com/words/source%20code) in order to [improve](https://www.wordnik.com/words/improve) its [readability](https://www.wordnik.com/words/readability), [reusability](https://www.wordnik.com/words/reusability) or [structure](https://www.wordnik.com/words/structure) without [affecting](https://www.wordnik.com/words/affecting) its [meaning](https://www.wordnik.com/words/meaning) or [behaviour](https://www.wordnik.com/words/behaviour)

Reply:
> Again, this does not look like a breaking change. This PR has "refactor" in the title. We can simply start using the new one because all the data contained in it is the same, no?

No, I renamed some fields to use inheritance, instead of duplicated code.

Old system just used few not related classes (for bedrock and java), and now I'm using inheritance for duplicated fields. This is the reason why `description` is `motd` now.

> Refactor - To [rewrite](https://www.wordnik.com/words/rewrite) existing [source code](https://www.wordnik.com/words/source%20code) in order to [improve](https://www.wordnik.com/words/improve) its [readability](https://www.wordnik.com/words/readability), [reusability](https://www.wordnik.com/words/reusability) or [structure](https://www.wordnik.com/words/structure) without [affecting](https://www.wordnik.com/words/affecting) its [meaning](https://www.wordnik.com/words/meaning) or [behaviour](https://www.wordnik.com/words/behaviour)

I'm fully following this definition:
- To [rewrite](https://www.wordnik.com/words/rewrite) existing [source code](https://www.wordnik.com/words/source%20code): old classes was fully removed and moved to its own file.
- in order to [improve](https://www.wordnik.com/words/improve) its [readability](https://www.wordnik.com/words/readability), [reusability](https://www.wordnik.com/words/reusability) or [structure](https://www.wordnik.com/words/structure): inheritance is more readable, reusadable, and structured than duplicated code.
- without [affecting](https://www.wordnik.com/words/affecting) its [meaning](https://www.wordnik.com/words/meaning) or [behaviour](https://www.wordnik.com/words/behaviour): goal of these classes the same as was. I also didn't touch the logic - constructors was rewrited, but working the same, as before.

Reply:
@PerchunPak is this still relevant after removing the deprecation?

Reply:
Looks like no.